src/ModelRegion.cpp

```diff
@@ -427,11 +436,80 @@ namespace geopm
         }
     }
 
+    TimedStreamModelRegion::TimedStreamModelRegion(double big_o_in,
+                                                   int verbosity,
+                                                   bool do_imbalance,
+                                                   bool do_progress,
+                                                   bool do_unmarked)
+        : StreamModelRegion(big_o_in, verbosity, do_imbalance, do_progress, do_unmarked)
+    {
+        m_name = "timed_stream";
+
+        if (big_o_in > 10.0) {
+            /// @todo: adjust to reuse same buffer when big-o would exhaust system memory. This depends on number of ranks.
```

Comment:
Is this future work?
Is there a future work issue open for this?

Reply:
Ideally we would do this for all regions types, not just this new one.  The current behavior is to die without explanation.  I'll open a new issue and remove this todo